ISLANDORA-2053: Update travis.yml to force PHP 5.3.3 to run under Ubuntu Precise

DiegoPino opened the pull request:
**ISLANDORA-2053**: (https://jira.duraspace.org/browse/ISLANDORA-2053)
http://irclogs.islandora.ca/2017-09-05.html
https://github.com/travis-ci/travis-ci/issues/2963
https://github.com/Islandora/islandora/pull/683

As of September 2017, Travis-CI has removed support for PHP 5.3 (and 5.2) for their Ubuntu Trusty Images, forcing us, in order  to keep Drupal's compatibility matrix tested, to run Travis-CI test for 5.3 on Ubuntu Precise


# What does this Pull Request do?

This pull Fixes our failing Travis-CI tests: 
Should allow PHP 5.3.3 to run under Ubuntu Precise and Higher versions under Ubuntu Trusty.
Changes the way we define which PHP versions and Fedora ones run in Travis

# What's new?
We moved PHP 5.3.3 version and the corresponding Fedora versions into the matrix section of the Travis YAML file. 


# How should this be tested?
 Travis-CI should pass and everything single PHP/FEDORA version combination should look good and very green.

# Interested parties
@Islandora/7-x-1-x-committers @nhart @jonathangreen @rosiel @ajstanley @bondjimbond @qadan

DiegoPino committed:
force PHP 5.3.3 to run under Ubuntu Precise

whikloj commented:
@DiegoPino was there a change to ImageMagick's `convert` in trusty too?
https://travis-ci.org/Islandora/islandora_solution_pack_pdf/jobs/275535579#L805

DiegoPino committed:
Tests: is gs there? imagemagick version?

DiegoPino committed:
Ghostscript not present, lets add it

Previous run said:  gs -v
The program 'gs' is currently not installed. To run 'gs' please ask
your administrator to install the package 'ghostscript’. So clearly we
need it installed. What i wonder if imagemagick will actually assume it
is there or i need to recompile

DiegoPino committed:
sudo

wonder why then “sudo” is even an option the global section… annoying..

DiegoPino commented:
@whikloj this one is passing now. Ghostscript comes not out of the box. Feel free to merge when every test is green. gracias
